### PR TITLE
Next next invoke config

### PR DESCRIPTION
**Pull request by ronalag**
1. Modified the config parser to parse invoke targets.
2. Modified the native packager to generate targets in the native packager.
3. Updated the version of the packager tools.
4. Added unit tests for invoke target.

The issue related to this feature is:

blackberry/BB10-Webworks-Packager#111

The packager tools dependency had to be updated so that the native packager would recognize the invoke target elements in the descriptor and produce correct invoke target entries in the bar manifest file. The version of the updated native packager is 1.5.

If you want to grab the updated packager tools you can do so by grabbing the artifact of the ci build for this feature branch and copying the dependencies folder of the artifact to the project root directory.

Note: The updated packager tools does not support the signing of applications. Therefore this pull request should not be merged until signing is supported by the signing authority.

### CODE REVIEW

**Comment by ronalag** [comment]
@EricLeiLi code review ?

File: lib/localize.js [hunk 1]
```diff
@@ -90,6 +90,12 @@ var Localize = require("localize"),
         },
         "EXCEPTION_FEATURE_DEFINED_WITH_WILDCARD_ACCESS_URI": {
             "en": "Invalid config.xml - no <feature> tags are allowed for this <access> element"
+        },
+        "EXCEPTION_INVOKE_TARGET_INVALID_ATTRIBUTES": {
+            "en": "Each invokeTarget element must specify a valid invokeType and invokeId attribute"
+        },
+        "EXCEPTION_INVOKE_TARGET_FILTER_INVALID": {
+            "en": "Each filter element must specify at least one action and one type"
```
Comment (tohman) [inline lib/localize.js]: One action and one _mime_-type is more clear.

**Comment by ronalag** [comment]
@EricLeiLi  The "- empty line + empty line" is there because I got rid of some extra whitespace in the native packager

**Comment by ronalag** [comment]
@jkeshavarzi I made updates based on our discussion. Can you review them?

**Comment by ronalag** [comment]
@jkeshavarzi the after each statement is necessary because some of the other tests parse different config.xml files in the test directory. Without this change some tests will be performed with the incorrect config.xml file. Having said that, I think that the after each should be a before each so I'll change the test case.

**Comment by jamesjhedley** [comment]
The config.xml object should really be a contant and should not change from test to test, requiring us to reset it after or before each test. It should remain pointed to the standard config.xml for most of our tests. The other cases where a different config is used ("should" really only be one [bare-minimum.xml]) we can use local variables to store the different location.

----- Original Message -----

**Comment by ronalag** [comment]
@jkesharvarshi made your recommended changes.

**Comment by EricLeiLi** [comment]
@jkeshavarzi Thanks for helping the code review;
@ishneur please proceed to the SV&V.

**Comment by jamesjhedley** [comment]
@EricLeiLi No problem.
@ronalag You should squash these commits before this branch is merged.

**Comment by jeffheifetz** [comment]
@ronalag Is the spec you're coding to documented anywhere, its not present in the issue you referenced.

**Comment by nukulb** [comment]
@jeffheifetz - it should on the wiki under invoke

**Comment by nukulb** [comment]
@ronalag - rebase from next

**Comment by ronalag** [comment]
@nukulb next or next-next?

**Comment by nukulb** [comment]
@ronalag - next-next

**Comment by ishneur** [comment]
For testing, is it sufficient to verify that correct information is added to blackberry-tablet.xml and to MANIFEST.MF, or should I also test the actual invocation of the WebWorks apps?

Also, I don't see a file called "bar-descriptor.xml". The file is called "blackberry-tablet.xml".

**Comment by nukulb** [comment]
You actually need to test invocation

**Comment by nukulb** [comment]
@ronalag - improve the commit message, please don't start all your commits with "made the follow changes" describe the change instead please. First line should always be a brief description.

**Pull request closed without merging**